### ISSUE TITLE
libnotify integration

### ISSUE THREAD

**Issue by ainola** [rawdigits/wee-slack#101]
For Linux users, it'd be great to get a notification on new messages.

**Comment by terlar** [comment]
You could use the weechat plugin: https://weechat.org/scripts/source/lnotify.py.html/

**Comment by ainola** [comment]
Sure, I had just noticed that 'growl integration' was on the roadmap. Since this plugin looks like it's trying to become a cohesive solution, it would be nice if libnotify was included.

If that's not the plan, please do close this bug. There's no reason to leave it open if there is no intention to support it.

**Comment by numkem** [comment]
I actually use this script and while it works great with bitlbee and "pure" IRC, with wee-slack I get a double notification.

I'm using dunst for notification daemon and this is what dust gets:
```
{
	appname: 'WeeChat notification'
	summary: 'infamous_derp'
	body: 'ok
ok'
	icon: 'info'
	category: 
	urgency: 0
	formatted: 'person: ok
ok'
	fg: #000000
	bg: #83c83d
	id: 3
	script: (null)
}
```

As you can see the body is duplicated on a new line.

**Comment by ainola** [comment]
I haven't been able to get any notification plugins to work with wee_slack at all, only regular IRC.

**Comment by mrrsm** [comment]
The lnotify plugin kind of works for me.  When someone messages me it will send a notification of the backlog from them as well which is kind of annoying but still works.  As long as I am not on the conversation it will notify me upon direct messages and items that get highlighted (I believe)

**Comment by keith** [comment]
I see the same thing @mrrsm mentioned sometimes as well. But it's definitely not 100%

**Comment by terlar** [comment]
It only sends you all messages if you didn't have the buffer open. I only get this issue if I close any of the chat buffers with people writing to me. By default weeslack will open these conversations every time. So the workaround is to not close the conversation buffers.

**Comment by mrrsm** [comment]
@terlar that is not a convenient option (for me) though.  

I believe that if you add previous messages to the backlog rather then the chat buffer you would get the appropriate messages sent to libnotify (or maybe it loads that for you, I am not really sure)

**Comment by rawdigits** [comment]
This may also be (mostly) fixed in #124 

I've been testing with my terrible-custom-growl-like system and I no longer get dupes when a DM buffer opens.

**Comment by tfheen** [comment]
Sounds like duplicates are fixed now, and we shouldn't have our own notification system (but integrate with those already out there, which we do), so closing this.
